syntax = "proto3";

package posts_analyser;

service PostsAnalyserDbController {
	rpc AddUserEntry(AddUserEntryRequest) returns (AddUserEntryResponse);
	rpc RemoveUserEntry(RemoveUserEntryRequest) returns (RemoveUserEntryResponse);
	rpc VerifyUserCredentials(VerifyUserCredentialsRequest) returns (VerifyUserCredentialsResponse);

  rpc AddPostEntry(AddPostEntryRequest) returns (AddPostEntryResponse);
	rpc ModifyPostEntry(ModifyPostEntryRequest) returns (ModifyPostEntryResponse);
	// rpc RemovePostEntry(RemovePostEntryRequest) returns (RemovePostEntryResponse);
	rpc GetPostEntry(GetPostEntryRequest) returns (GetPostEntryResponse);

	rpc GetPostsEntries(GetPostsEntriesRequest) returns (GetPostsEntriesResponse);
}

message AddUserEntryRequest {
  string username = 1;
  string password = 2;
}

message AddUserEntryResponse {}

message RemoveUserEntryRequest {
  string username = 1;
}

message RemoveUserEntryResponse {}

message VerifyUserCredentialsRequest {
  string username = 1;
  string password = 2;
}

message VerifyUserCredentialsResponse {}

message AddPostEntryRequest {
  string title = 1;
  string author = 2;
  string content = 3;
}

message AddPostEntryResponse {}

message ModifyPostEntryRequest {
  int32 id = 1;
  string title = 2;
  string content = 3;
  string classification = 4;
}

message ModifyPostEntryResponse {}

// message RemovePostEntryRequest {
//  int32 id = 1;
// }

// message RemovePostEntryResponse {}

message GetPostEntryRequest {
  int32 id = 1;
}

message GetPostEntryResponse {
  PostEntry post_entry = 1;
}

message PostEntry {
  int32 id = 1; 
  string title = 2;
  string author = 3;
  string content = 4;
  string created_at = 5;
  string modified_at = 6;
  string tagged_at = 7;
  string classification = 8;
}


message GetPostsEntriesRequest {
}

message GetPostsEntriesResponse {
  repeated PostEntry post_entries = 1;
}
